ql/termstructures/yield/zeroyieldstructure_simple.hpp

```diff
@@ -0,0 +1,103 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2000, 2001, 2002, 2003 RiskMap srl
+ Copyright (C) 2003, 2004 StatPro Italia srl
+ Copyright (C) 2009 Ferdinando Ametrano
+ Copyright (C) 2019 SoftSolutions! S.r.l.
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file zeroyieldstructure.hpp
+    \brief Zero-yield based term structure
+*/
+
+#ifndef quantlib_zero_yield_structure_simple_hpp
+#define quantlib_zero_yield_structure_simple_hpp
+
+#include <ql/termstructures/yieldtermstructure.hpp>
+
+namespace QuantLib {
+
+//! Zero-yield term structure
+/*! This abstract class acts as an adapter to YieldTermStructure
+    allowing the programmer to implement only the
+    <tt>simpleZeroYieldImpl(Time)</tt> method in derived classes.
+
+    Discount and forward are calculated from simple zero yields.
+
+    Zero rates are assumed to be annual simple compounding.
+
+    \ingroup yieldtermstructures
+*/
+class ZeroYieldStructureSimple : public YieldTermStructure {
```

Comment:
I'd go for grammar and call it `SimpleZeroYieldStructure`, but it might be read as a simple structure of zero rates instead of a structure of simply-compounded zero rates.

In any case, since at this time the only class inheriting this is `InterpolatedZeroCurveSimple`, I'm kind of thinking we could avoid this one for the time being and implement `discountImpl` in `InterpolatedZeroCurveSimple` directly.  (The idea would be to avoid proliferation of structures until we think, e.g., if we have a way to parameterize zero-structures on compounding.)

Reply:
That's fine as well. Would you have time to do these changes? That would just be great.

Reply:
Yes, at some point 😄

Reply:
Cool thanks 👍